### PR TITLE
add top level module documentation for Sipity with some use cases

### PR DESCRIPTION
**Pull request by no-reply**
this is an attempt to document Sipity at the top level. the focus here is on
describing the domain model and explaining some of the use cases. i think a lot
more is required to make this crystal clear; e.g. example code, state diagrams,
and a deep description of Roles, Notifications and Methods is probably called
for.

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: app/models/sipity.rb [hunk 1]
```diff
@@ -2,6 +2,36 @@
...
+# authorized to carry out each action.
+#
+# Any uniquely identifiable object can be managed by a {Sipity::Workflow}.
+# Normally Hyrax uses workflows to handle the deposit process and maintenance
+# lifecycle of repository objects at the level of the Work (within the Hydra
+# Works model). Objects are represented within the Sipity engine's domain model
+# by a {Sipity::Entity}. Each object has at most one {Sipity::Entity}, is
+# governed by one {Sipity::Workflow}, and in one {Sipity::WorkflowState} at any
+# given time.
+#
```
Comment (lsitu) [inline app/models/sipity.rb:21]: Do we want to mention Sipity::EntitySpecificResponsibility and workflow comments feature at this top level?

**Review by lsitu — APPROVED** [review]
@no-reply Looks great! Thanks. I just added a little question.

File: app/models/sipity.rb (hunk 1 above)
Comment (no-reply) [reply app/models/sipity.rb:21]: yeah. i think so. i'm just starting to understand how these fit in, and haven't yet figured out how to document them. do you have ideas?

Comment (lsitu) [reply app/models/sipity.rb:21]: I don't know what's the best way to address it either but it's a feature that we may interest in for Comet workflow. Maybe we can add a use case for it?

**Pull request merged**